File: PA/tema_gennaio/es4_ereditarieta_java/src/VeicoloElettrico.java
public class VeicoloElettrico extends Veicolo implements Alimentazione {

    private float wBatteria;

    public VeicoloElettrico(float batteria) {
        this.wBatteria = batteria;
    }

    public float getBatteria() {
        return wBatteria;
    }

    // mia assunzione, il veicolo elettrico percorre 5 km con 1 punto percentuale di batteria

    @Override
    public void percorri(float km) {
        if ((wBatteria - km/5) >= 0){
            this.km = km;
            wBatteria = wBatteria - km/5;
        }else{
            System.out.println("Batteria insufficiente");
        }
       
    }

    @Override
    public void rifornisci(float quantita) {
        wBatteria += quantita;
    }

} 

File: PA/tema_gennaio/es4_ereditarieta_cpp/VeicoloElettrico.cpp
//
// Created by mitic on 25/01/2025.
//
#include "Veicolo.cpp"

class VeicoloElettrico : public Veicolo{
  private:
    float wBatteria;
    public:
    VeicoloElettrico(float batteria){
      this->wBatteria = batteria;
    }
};


